plugins/scaffolder-backend-module-confluence-to-markdowwn/src/actions/confluence/confluenceToMarkdown.ts

```diff
@@ -0,0 +1,182 @@
+          targetPath: dirPath,
+        },
+      });
+
+      for (let p = 0; p < confluenceUrls.length; p++) {
+        const confluenceUrl = confluenceUrls[p];
+
+        const { spacekey, title, titleWithSpaces } =
+          await createConfluenceVariables(confluenceUrl);
+
+        // This calls confluence to get the page html and page id
+        const getConfluenceDoc = await fetchConfluence(
+          `/rest/api/content?title=${title}&spaceKey=${spacekey}&expand=body.export_view`,
+          config,
+        );
+        if (getConfluenceDoc.results.length === 0) {
+          throw new InputError(
+            `Could not find document ${confluenceUrl}. Please check your input.`,
+          );
+        }
+        // This gets attachements for the confluence page if they exist
+        const getDocAttachments = await fetchConfluence(
+          `/rest/api/content/${getConfluenceDoc.results[0].id}/child/attachment`,
+          config,
+        );
+
+        if (getDocAttachments.results.length) {
+          fs.mkdirSync(`${dirPath}/${filePathToMkdocs}docs/img`, {
+            recursive: true,
+          });
+          productArray = await getAndWriteAttachments(
+            getDocAttachments,
+            dirPath,
+            config,
+            filePathToMkdocs,
+          );
+        }
+
+        ctx.logger.info(
+          `starting action for converting ${titleWithSpaces} from Confluence To Markdown`,
+        );
+
+        // This reads mkdocs.yml file
+        const mkdocsFileContent = await readFileAsString(repoFileDir);
+        const mkdocsFile = parseFromYaml(mkdocsFileContent);
+        ctx.logger.info(
+          `Adding new file - ${titleWithSpaces} to the current mkdocs.yml file`,
+        );
+
+        // This modifies the mkdocs.yml file
+        if (mkdocsFile !== undefined && mkdocsFile.hasOwnProperty('nav')) {
+          const { nav } = mkdocsFile;
+          if (!nav.some((i: Obj) => i.hasOwnProperty(titleWithSpaces))) {
+            nav.push({
+              [titleWithSpaces]: `${titleWithSpaces.replace(/\s+/g, '-')}.md`,
+            });
+            mkdocsFile.nav = nav;
+          } else {
+            throw new ConflictError(
+              'This document looks to exist inside the GitHub repo. Will end the action.',
+            );
+          }
+        }
+        const newTemplateFileContent = transformToYaml(mkdocsFile);
+        await updateFile(newTemplateFileContent, repoFileDir);
+
+        // This grabs the confluence html and converts it to markdown and adds attachments
+        const html = getConfluenceDoc.results[0].body.export_view.value;
+        const markdownToPublish = NodeHtmlMarkdown.translate(html);
+        let newString: string = markdownToPublish;
+        productArray.map((product: string[]) => {
+          const regex = product[1].includes('pdf')
+            ? new RegExp(`(\\[.*?\\]\\()(.*?${product[0]}.*?)(\\))`, 'gi')
```

Comment:
Is there something more structured that's available from t he confluence API to not have to resort to this regexes 😅

Reply:
The issue here is that when we pull the html from confluence, the reference to the images are still in confluence and not in the GitHub repo. So the intense regex is for removing that reference to confluence and updating it to look for the images/pdfs/etc saved in the `/img` folder. It is something I did internally so users don't have to go back and update it manually but if it is too much(because it is a lot), I can remove this step.

Reply:
I'm all for helping the users. I suggest adding a comment showing the URLs here to help any future developer trying to understand this regex.

Reply:
done